src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -184,6 +185,8 @@ public Task Stop()
                         }
                         finally
                         {
+                            HttpContextFactory.Dispose(httpContext);
```

Comment:
This needs to happen after `FireOnStarting` and `FireOnCompleted`. Both these methods execute user code.